### ISSUE TITLE
Prevent vaccine waste by notifying vaccination volunteers

### ISSUE THREAD

**Issue by manuliner** [corona-warn-app/cwa-wishlist#341]
## Feature description

Prevent vaccine waste by notifying vaccination volunteers

1. Users can agree to receive a notification if vaccine needs to be used up.

2. Vaccination centers can enter the quantity and expiration period via their account.

3. The system uses a bit of IT voodoo to determine the next vaccination volunteers. They then have X amount of time to respond. According to the principle of first come, first serve, those who have responded the fastest get the commitments. In the confirmation you will find the address, OTP and the deadline for the vaccinated person to come to the center. Otherwise the next request will be sent out.

## Problem and motivation

Once prepared, the COVID-19 vaccine has a shelf life of only a few hours and must be administered during this period.

If no vaccine candidate is found, the vaccine is discarded.

Even with the very first COVID-19 vaccination in Germany, 7 vaccine doses remained at the end of the day.

Fortunately, people willing to vaccinate could be found quickly from the vaccination team.
Currently, for example, surrounding hospitals and their staff are also supplied with the surplus vaccine.

However, it is foreseeable that additional persons willing to vaccinate spontaneously will be needed to avoid disposal of the vaccine.

This is where a reporting function in the Corona app would be helpful as a vaccinator.

Currently, I am trying to establish contacts with vaccination centers and public health departments in NRW to get a public position on the procedure with leftover vaccine doses. 

However, my research so far shows that in every case vaccine doses have been left over, but then distributed to employees of care facilities and surrounding hospitals. However, this resource of vaccination volunteers is limited. (All my friends in this area have already been vaccinated). I am trying to get some numbers and relevant Statistics to this.

## Is this something you're interested in working on

I wont' be able to code, as this is an ASAP Feature and i am not familiar with mobile dev. But i will in support in Product Owning and PM

**Comment by MikeMcC399** [comment]
@manuliner 
It sounds like your proposal would need to know the user's location in order to be effective, however the Corona-Warn-App does not know the device's location and is forbidden to use it. (See for instance https://www.coronawarn.app/assets/documents/cwa-privacy-notice-en.pdf Section 5). Did I understand your proposal correctly?

"The app’s entire system has been programmed to process as little personal data as possible. This means that, when you use exposure logging, warn other users, or retrieve a test result, the system does not collect any data that would allow the RKI or other users to infer your identity, your name, your location or other personal details."

The organisation of vaccinations is being handled by each State according to the information on https://www.zusammengegencorona.de/. I'm not in NRW, so I'm not following how it is being done there. I do know though that in Bavaria, where I am located, that anybody can sign up with an account on http://impfzentren.bayern/ to be informed by e-mail / SMS when they can come to be vaccinated. It asks a lot of questions starting with your date-of-birth, medical conditions, including your environment (such as the type of work you do, if applicable, etc.). It uses all that information to prioritise the order of appointments. A system like that could perhaps be extended to offer short-term vaccination appointments.

**Comment by manuliner** [comment]
Yes there will be a need for a rough location ( city would be enough, to determine the next "Impfzentrum"). So probably  CWA will be not the right place for this functionality.


@MikeMcC399  thank you for the Input. I am already contacting all Vaccaination Administration regarding this issue. There is no need to develop a new Application for this, as long as the State has one already running. But they should consider this Feature to solve the Issue. I mean, even if the automatically inform already registered People over there Database, that an short term appointment will be availible, would make a huge difference.


As is did found out yesterday, in Rheinland Pfalz they where forced to throw 228 Vaccinces into the rubbish, since 27.12.2020 

https://www.rheinpfalz.de/lokal/kreis-germersheim_artikel,-impfzentrum-w%C3%B6rth-noch-keine-impfdosis-weggeworfen-_arid,5158876.html?reduced=true

![e9451a54-48c0-48c5-90d5-59d2089da477](https://user-images.githubusercontent.com/5088151/105576193-6610d980-5d71-11eb-8aff-ba00ee8d38d6.jpg)

**Comment by tyll** [comment]
Is there now a solution for this problem for interested doctors? The inefficient communication for appointments is still a problem in Germany, anecdotal evidence: https://twitter.com/hodisplace/status/1461656270629482501

**Comment by manuliner** [comment]
Yes after @MikeMcC399  answer we started www.impfbruecke.de to support official vaccination centres nearly free of charge. Later there where others which made an approach for doctors www.sofortimpfen.de. We didn't serve doctors with impfbrücke due the realy high ammount of customer support. So basically it is an money problem. Most doctors who where needing an appointment system can actually get them at calendly, doctolib and jameda. But they don't want to pay for it. Impfbrücke and sofortimpfen where services who ran on really low budget.  After this budget was used we are not able to provide this service anymore.  If you take money to keep it running, no one will use it.

I still believe cwa would be the best place for a "centralised"  appointment system. And there are ways to do it. Sofort Impfen has proven it.

At the moment we are thinking about collecting money to start impfbruecke again but it would be nicer to use this money to implement this kind of functionality in cwa.

Open API for every appointment system as jameda,  doctolib and the official ones to use cwa as an appointment finder in your surrounding area

If someone at cwa is willing to think about a approach impfbrücke will be happy to support with the experience of the last year in this field. An I believe that the persons from sofortimpfen will be in board as well

**Comment by tyll** [comment]
Is Impfbrücke open-source software? What services need to be paid to get it running?

**Comment by manuliner** [comment]
currently Impfbruecke is not opensource, but i am willingly to do change it but i must discuss this topic with the other teammembers.
sofort-impfen is opensource (https://github.com/kiebitz-oss/)

Technical Services: 

- As Impfbrucke  is an service using SMS the cost of theses needs to be covered ( apporx 0,70€ to 5 € per appointment, depending on the database).  
-  Servers

Human Services:

- During the vaccination campaign this year the management of all the vaccination centers and bugfixing was a fulltime job for 3 People.
